src/block/SweetBerryBush.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\block;
+
+use pocketmine\block\utils\BlockDataSerializer;
+use pocketmine\event\block\BlockGrowEvent;
+use pocketmine\item\Fertilizer;
+use pocketmine\item\Item;
+use pocketmine\item\VanillaItems;
+use pocketmine\math\Facing;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
+use pocketmine\world\BlockTransaction;
+use function mt_rand;
+
+class SweetBerryBush extends Flowable {
+
+	protected $age;
+
+	protected function writeStateToMeta() : int{
+		return $this->age;
+	}
+
+	public function readStateFromData(int $id, int $stateMeta) : void{
+		$this->age = BlockDataSerializer::readBoundedInt("age", $stateMeta, 0, 7);
+	}
+
+	public function getStateBitmask() : int{
+		return 0b111;
+	}
+
+	public function getAge() : int{ return $this->age; }
+
+	/** @return $this */
+	public function setAge(int $age) : self{
+		if($age < 0 || $age > 7){
+			throw new \InvalidArgumentException("Age must be in range 0-7");
+		}
+		$this->age = $age;
+		return $this;
+	}
+
+	public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		switch ($blockReplace->getSide(Facing::DOWN)->getId()){
+
+			case BlockLegacyIds::GRASS:
+			case BlockLegacyIds::DIRT:
+			case BlockLegacyIds::PODZOL:
+				return Block::place($tx, $item, $blockReplace, $blockClicked, $face, $clickVector, $player);
+
+			default:
+				return false;
+
+		}
+	}
+
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		if ($item instanceof Fertilizer) {
+			$block = clone $this;
+			$block->age++;
+			if ($block->age > 7) {
+				$block->age = 7;
+
+			}
+
+			$ev = new BlockGrowEvent($this, $block);
+			$ev->call();
+
+			if (!$ev->isCancelled()) {
+				if ($ev->getNewState()->getAge() >= 4) {
+					$ev->getNewState()->setAge(1);
```

Comment:
I'm not sure what this is doing, but it will prevent plugins from setting the age to anything higher than 4.